src/display.jl

```diff
@@ -241,3 +227,12 @@ superscript(i::Integer) = map(repr(i)) do c
     c == '0' ? '\u2070' :
     error("unexpected character")
 end
+
+Base.print(io::IO, x::Quantity) = show(io, "text/plain", x)
+Base.print(io::IO, x::Dimension) = show(io, "text/plain", x)
+Base.print(io::IO, x::Unit) = show(io, "text/plain", x)
+Base.print(io::IO, x::Union{StepRange{T},StepRangeLen{T}}) where T<:Quantity = show(io, "text/plain", x)
+Base.print(io::IO, x::typeof(NoDims)) = show(io, "text/plain", x)
+Base.print(io::IO, x::Unitlike) = show(io, "text/plain", x)
+Base.print(io::IO, x::MixedUnits{T,U}) where {T,U} = show(io, "text/plain", x)
+Base.print(io::IO, x::FreeUnits{N,D,A}) where {N,D,A} = show(io, "text/plain", x)
```

Comment:
```suggestion
Base.print(io::IO, x::Unitlike) = show(io, "text/plain", x)
Base.print(io::IO, x::MixedUnits) = show(io, "text/plain", x)
Base.print(io::IO, x::LogScaled) = show(io, "text/plain", x)
Base.print(io::IO, x::IsRootPowerRatio) = show(io, "text/plain", x)
```
* `NoDims` and `FreeUnits` are covered by `Unitlike`
* the type parameters in the `MixedUnits` method are not needed
* methods for `LogScaled` and `IsRootPowerRatio` are added